aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -2799,6 +2800,8 @@ std::tuple<Tensor&, Tensor&> linalg_eig_out(const Tensor& input, Tensor& values,
 }
 
 std::tuple<Tensor, Tensor> linalg_eig(const Tensor& input) {
+  TORCH_CHECK(input.isfinite().all().item<bool>(), "linalg_eig: input tensor should not contain infs or NaNs.");
```

Comment:
`linalg_eig` functions calls to `linalg_eig_out`. Could you please move this check to the `linalg_eig_out` function?